src/sid/update_states.py

```diff
@@ -161,11 +161,12 @@ def _update_info_on_new_tests(
     # For everyone who received a test result, the countdown for the test processing
     # has expired. If you have a positive test result (received_test_result &
     # immune) you will leave the state of knowing until your immunity expires.
-    states["new_known_case"] = states["received_test_result"] & states["immunity"] > 0
+    states["new_known_case"] = states["received_test_result"] & (
+        states["infectious"] | states["symptomatic"]
+    )
     states.loc[states["new_known_case"], "knows_immune"] = True
-    states.loc[states["new_known_case"], "cd_knows_immune_false"] = states.loc[
-        states["new_known_case"], "cd_immune_false"
-    ]
+    states.loc[states["new_known_case"], "cd_knows_immune_false"] = 1000  # Needs to be
+    # checked TODO: # noqa: T000
```

Comment:
This needs to be checked.

Reply:
I think that line is not needed anymore because we don't use that countdown for anything if I see it right. We don't use it in sid-germany and I don't think we use it elsewhere in sid.